docs/resource/README.md

```diff
@@ -105,7 +105,50 @@ as specified in the [Resource SDK specification](https://github.com/open-telemet
 
-**[2]:** MUST be unique for each instance of the same `service.namespace,service.name` pair (in other words `service.namespace,service.name,service.instance.id` triplet MUST be globally unique). The ID helps to distinguish instances of the same service that exist at the same time (e.g. instances of a horizontally scaled service). It is preferable for the ID to be persistent and stay the same for the lifetime of the service instance, however it is acceptable that the ID is ephemeral and changes during important lifetime events for the service (e.g. service restarts). If the service has no inherent unique ID that can be used as the value of this attribute it is recommended to generate a random Version 1 or Version 4 RFC 4122 UUID (services aiming for reproducible UUIDs may also use Version 5, see RFC 4122 for more recommendations).
+**[2]:** MUST be unique for each instance of the same `service.namespace,service.name` pair
+(in other words `service.namespace,service.name,service.instance.id` triplet MUST be globally unique).
+The ID helps to distinguish instances of the same service that exist at the same time
+(e.g. instances of a horizontally scaled service). It is preferable for the ID to be persistent
+and stay the same for the lifetime of the service instance, however it is acceptable that
+the ID is ephemeral and changes during important lifetime events for the service
+(e.g. service restarts).
+If the service has no inherent unique ID that can be used as the value of this attribute
+it is recommended to generate a random Version 1 or Version 4
+[RFC 4122](https://www.ietf.org/rfc/rfc4122.txt) UUID. Services aiming for reproducible UUIDs may also
+use Version 5. UUIDs are typically recommended, as we only need an opaque yet reproducible value for
+the purposes of identifying a service instance. Similar to what can be seen in the man page for the
+`/etc/machine-id` file, the underlying data, such as pod name and namespace should be treated as
+confidential by this algorithm, being the user's choice to expose it or not via another resource attribute.
+When a UUID v5 is generated, the UUID's namespace MUST be set to:
+`${telemetry.sdk.name}.${telemetry.sdk.language}.${service.name}`, so that the same service yields the
+same ID if the same value (`host.id`, `/etc/machine-id`, and so on) remains the same. It would still
+yield different results for different services on the same host.
+
+SDKs are required to use the following algorithm when generating `service.instance.id`:
+
+- If the user has provided a `service.instance.id`, via environment
+  variable, configuration or custom resource detection, this will
+  always be used and honored over generated IDs.
+- When any of the below combinations of resource attribute are provided, they should be used as the input
+  for generating a UUID v5. The values within each combination should be separated with dots:
+  * `container.id`
+  * `k8s.namespace.name`/`k8s.pod.name`/`k8s.container.name`
```

Comment:
`k8s.pod.name` will remain same for statefulset pods across recreates, but with other k8s controllers (deployment, cronjobs, etc) the pod name changes. So the uuid for statefulset pods will be the same with this combination even after a recreate. With other options in this section, the uuid does change when underlying host changes, except for statefulset pod. Should we make the change in uuid more consistent for k8s pods by using the `k8s.pod.uid` here?

Reply:
> So the uuid for statefulset pods will be the same with this combination even after a recreate

That's on purpose: on a previous discussion with @jsuereth, we agreed that we understand different restarts of stateful and daemon sets to be the same instance, similar to a process being restarted on the same machine (which would be using the /etc/machine-id in this logic).